File: validation/validation-annotation-processor/src/test/java/ru/tinkoff/kora/validation/annotation/processor/testdata/ValidationLifecycle.java
package ru.tinkoff.kora.validation.annotation.processor.testdata;

import ru.tinkoff.kora.annotation.processor.common.MockLifecycle;
import ru.tinkoff.kora.validation.Validator;

public record ValidationLifecycle(Validator<Foo> foo, Validator<Bar> bar) implements MockLifecycle {}

File: validation/validation-symbol-processor/src/test/kotlin/ru/tinkoff/kora/validation/symbol/processor/testdata/ValidationLifecycle.kt
package ru.tinkoff.kora.validation.symbol.processor.testdata

import ru.tinkoff.kora.annotation.processor.common.MockLifecycle
import ru.tinkoff.kora.validation.Validator

data class ValidationLifecycle(val foo: Validator<Foo>, val bar: Validator<Bar>) : MockLifecycle
